share/spice/sentiment_analysis/document/sentiment_analysis_document.js

```diff
@@ -0,0 +1,77 @@
+var ddg_spice_sentiment_analysis = {
```

Comment:
>So, $.getJSON and item.set('property', value) should be used only if JSONP is not available by the API provider. Is this correct?

Not quite -- seeing as we're dealing with AJAX here, the JSONP isn't really important. A JSON endpoint can be requested, and we can use an inline callback function to handle the result, instead of defining another global function for a second JSONP call. 

`item.set()` doesn't have any connection to JSON or JSONP, it's used to update the IA data after rendering, and to force the IA to re-render with newly updated info. 

>Since the API I'm using supports JSONP, I'm adding &callback={{callback}} to the request's URL and not setting wrap_jsonp_callback to 1. When I hit the second API with $.getScript(..) , I parse the results and add it to the object that will be passed as argument of the function render() in order to all the data to be rendered. Am I doing it properly?

I would suggest you drop `&callback=` and keep `wrap_jsonp_callback` disabled. Use `getJSON` and deal with your data in a local callback. You'll want to cleanup any data and call `item.set()` to update the UI with your new data.

I would do all this work inside an `onItemShown()` event.

https://github.com/duckduckgo/zeroclickinfo-spice/blob/71cfc655bf5ea89243c812ec35f23e1d1fbea6af/share/spice/news/news.js#L112 -- this is a good example of how you should structure your code (i.e. all in a closure, only 1 callback exposed to the DOM, truly private helper vars and functions defined inside closure, but outside callback definition

Does this make more sense now?